### PR TITLE
removed legacy methods

### PR DESCRIPTION
**Pull request by DavoII**
https://github.com/SocialMediaEmotions/yeppik/issues/1404

### CODE REVIEW

File: src/SortableList.js [hunk 1]
```diff
@@ -74,7 +74,12 @@ export default class SortableList extends Component {
     scrollEnabled: this.props.scrollEnabled
   };
 
-  componentWillMount() {
+  constructor(props) {
+    super(props);
+    this._resolveLayout();
```
[the remark this answers is not in the document]
Comment (DavoII) [reply src/SortableList.js]: In my understanding we need layout set before rendering, `componentDidMount ` works after it and it does not rerender anything, so If I change to it it won't render anything.

[the remark this answers is not in the document]
Comment (kukiron) [reply src/SortableList.js]: For the side effects, use `componentDidMount`.
Besides, `state` & other values should be initialized inside `constructor`, which isn't in this case. They're initialized outside.

[the remark this answers is not in the document]
Comment (kukiron) [reply src/SortableList.js]: For the side effects, use `componentDidMount`.
Besides, `state` & other values should be initialized inside `constructor`, which isn't in this case. They're initialized outside.

If you use `constructor` here, the code should be like this - 
```js
export default class SortableList extends Component {
  ...
  ...
  ...

  constructor(props) {
    super(props);    
    this._rows = {};
    /**
     * Stores promises of rows’ layouts.
     */
    this._rowsLayouts = {};
    this._resolveRowLayout = {};
    this._contentOffset = {x: 0, y: 0};

    this.state = {
      animated: false,
      order: this.props.order || Object.keys(this.props.data),
      rowsLayouts: null,
      containerLayout: null,
      data: this.props.data,
      activeRowKey: null,
      activeRowIndex: null,
      releasedRowKey: null,
      sortingEnabled: this.props.sortingEnabled,
      scrollEnabled: this.props.scrollEnabled
    };
  }

  componentDidMount() {
    this.state.order.forEach((key) => {
      this._rowsLayouts[key] = new Promise((resolve) => {
        this._resolveRowLayout[key] = resolve;
      });
    });

    if (this.props.renderHeader && !this.props.horizontal) {
      this._headerLayout = new Promise((resolve) => {
        this._resolveHeaderLayout = resolve;
      });
    }
    if (this.props.renderFooter && !this.props.horizontal) {
      this._footerLayout = new Promise((resolve) => {
        this._resolveFooterLayout = resolve;
      });
    }
  }

  ...
  ...
  ...
}
```

[the remark this answers is not in the document]
Comment (DavoII) [reply src/SortableList.js]: You're right about state in `constructor` I'll change it to be initialized in it, but removing `_resolveLayout` will cause layout not to render, so I'll add `_resolveLayout` to `componentDidMount ` too

[the remark this answers is not in the document]
Comment (DavoII) [reply src/SortableList.js]: Hey, I just checked again for rendering and for some odd reason it does render I don't know what is the difference when I tried days ago, but l'll change it now:D

[the remark this answers is not in the document]
Comment (DavoII) [reply src/SortableList.js]: Sorry about that, did the changes you requested

**Pull request merged**